example/provider-local/garden/local/kustomization.yaml

```diff
@@ -6,5 +6,6 @@ resources:
 - controllerregistration.yaml
 
 patches:
+# patch-controllerdeployment-prow.yaml must be enlisted first because it contains an empty values map when running in a non-prow environment.
```

Comment:
I have second thoughts on whether this is the right fix for the problem.
Maybe we should rather fix the patch generation to not generate a patch that removes the entire `values` map instead.
If the base manifest had other values, using a specific patch order wouldn't suffice.

What about the following idea?
If not running in prow, we add a `{}` value to the patch so that the patch doesn't remove any existing values. For example: https://github.com/gardener/gardener/compare/master...timebertt:gardener:fix-patch-prow

With this change, the image vector overwrites are not removed anymore:
```
$ k kustomize example/provider-local/garden/local | yq 'select(.kind=="ControllerDeployment") | .helm.values'
imageVectorOverwrite: |
  images:
  - name: machine-controller-manager-provider-local
    repository: garden.local.gardener.cloud:5001/local-skaffold_machine-controller-manager-provider-local
    tag: v1.111.0-dev-ad20d21b2b4e0433b714e9250f774b56c05235b461b6e386c7fd7cca48fccbd2
```

Reply:
yes, looks good 🚀